src/pyenphase/models/system_production.py

```diff
@@ -42,12 +42,12 @@ def from_production(cls, data: dict[str, Any]) -> EnvoySystemProduction:
         inverters = find_dict_by_key(all_production, "inverters")
 
         # This is backwards compatible with envoy_reader
+        # envoy metered without configured CT has whLifetime and wNow in inverters
+        # whLastSevenDays and whToday are incorrect for both so either can be used
         now_source = eim if eim["activeCount"] else inverters
 
         return cls(
-            watt_hours_lifetime=int(
-                round(eim.get("whLifetime") or inverters.get("whLifetime") or 0)
-            ),
+            watt_hours_lifetime=int(round(now_source["whLifetime"])),
```

Comment:
This could use a comment about why we use the `now_source` for `watt_hours_lifetime`

Reply:
It has the comment right above now_source calculation already.

```python
        # This is backwards compatible with envoy_reader
        # envoy metered without configured CT has whLifetime and wNow in inverters
        # whLastSevenDays and whToday are incorrect for both so either can be used
        now_source = eim if eim["activeCount"] else inverters
```